tests/tck-build-logic/src/main/java/org/graalvm/internal/tck/DockerTask.java

```diff
@@ -0,0 +1,74 @@
+package org.graalvm.internal.tck;
+
+import org.gradle.api.DefaultTask;
+import org.gradle.api.tasks.TaskAction;
+import org.gradle.api.tasks.options.Option;
+
+import java.io.*;
+import java.util.HashSet;
+import java.util.Set;
+
+public class DockerTask extends DefaultTask {
+
+    private String coordinates;
+
+    @Option(option = "coordinates", description = "Coordinates in the form of group:artifact:version")
+    void setCoordinates(String coordinates) {
+        this.coordinates = coordinates;
+    }
+
+
+    @TaskAction
+    void run() throws IOException, IllegalStateException{
+
+        Coordinates coordinates = Coordinates.parse(this.coordinates);
+        String coordinatesTestRoot = getProject().file(CoordinateUtils.replace("tests/src/$group$/$artifact$/$version$", coordinates)).getPath();
+        File dockerList = new File(coordinatesTestRoot + "/docker.txt");
+
+        if (!dockerList.exists()) {
+            // TODO once we have a docs for this issue, add a link to it in this error message
+            System.out.println("Docker file not exist.");
+            return;
+        }
+
+        Set<String> allowedImages = getAllowedImages();
+        BufferedReader isr = new BufferedReader(new InputStreamReader(new FileInputStream(dockerList)));
```

Comment:
Use `Files.readLines()` instead.

Reply:
Even better, inject `ExecOperations` and avoid use of the system exec API which is known to cause issues in Gradle, in particular under Windows. This API (`ExecOperations`) is designed precisely to fork processes and capture their output.